packages/ember-glimmer/lib/utils/references.js

```diff
@@ -38,6 +38,19 @@ export const UPDATE = symbol('UPDATE');
 
 export { NULL_REFERENCE, UNDEFINED_REFERENCE } from '@glimmer/runtime';
 
+let maybeFreeze = () => {};
+if (DEBUG) {
+  maybeFreeze = (obj) => {
```

Comment:
Can you add a note here regarding intended usage? I notice you're left the `if (DEBUG) {` statements even though the function is a noop when `!DEBUG` and I presume it is for DCE to do its thing. A note here might help a future traveller from thinking they can rely on the noop in non-debug.